test/test_misc.py

```diff
@@ -2152,6 +2153,73 @@ def f3(x):
             f3(torch.ones(6))
         self.assertEqual(cnt.frame_count, 0)
 
+    @requires_static_shapes
+    def test_shape_change_recompile_permissive(self):
+        from torchdynamo.eval_frame import PartiallyDynamicTensor
```

Comment:
eval_frame.py is the wrong place for this

Reply:
100% - this is just a random place for the sake of the prototype/RFC - I will go to its own file.